extract/utils.py

```diff
@@ -103,4 +103,5 @@ def execute_dataframe(engine, query):
         return df
     except Exception as e:
         return print(f"Oh no! There was an error executing your query: {e}")
-    cur.close()
```

Comment:
Most likely this will happen because `results` will be null.

IMHO we should change the function so that:
- it won't catch the exception but rather leave it to the caller
- it doesn't close the engine.

Making these changes would require adjusting all the code that is using this function. This would be a rather big PR as the function is used all over the repo. I'd first finish the round of tests on all python code and then make this change, just to have a bit more confidence from the tests.